### PR TITLE
Support persisting collapsible section state

### PR DESCRIPTION
**Pull request by DarpanKulkarni**
Added an option that allow to persist the sections collapse state to the browser's LocalStorage. This enhancement leverages the Alpine.js Persist plugin, which is now included with Livewire 3.

The sections state can be persisted using `->persistCollapseState()`.

The localStorage keys use the component's id, which is generated using the section heading. If, for some reason, two or more sections have the same headings, the `->id()` can be used to avoid conflicts.

The changes have been throughly tested. However, I have identified one minor issue related to flash of content during the page reload when the section is persisted and expanded. This is because the collapse toggle behaviour is managed by Alpine.js on the client side, and it's not feasible to conditionally add x-cloak.  I think this is minor issue an can be ignored. I will continue investigating potential solutions to address this issue, any suggestions or feedback from the team are welcomed and highly appreciated.

Thank you for your time and consideration.

### CODE REVIEW

File: packages/infolists/src/Components/Concerns/CanBeCollapsed.php [hunk 1]
```diff
@@ -6,11 +6,13 @@
 
 trait CanBeCollapsed
 {
-    protected bool | Closure $isCollapsed = false;
+    protected bool|Closure $isCollapsed = false;
```
Comment (saade) [inline packages/infolists/src/Components/Concerns/CanBeCollapsed.php]:
```suggestion
    protected bool | Closure $isCollapsed = false;
```

File: packages/infolists/src/Components/Concerns/CanBeCollapsed.php [hunk 2]
```diff
@@ -6,11 +6,13 @@
...
 
-    protected bool | Closure | null $isCollapsible = null;
+    protected bool|Closure|null $isCollapsible = null;
```
Comment (saade) [inline packages/infolists/src/Components/Concerns/CanBeCollapsed.php]:
```suggestion
    protected bool | Closure | null $isCollapsible = null;
```

File: packages/infolists/src/Components/Concerns/CanBeCollapsed.php [hunk 3]
```diff
@@ -6,11 +6,13 @@
...
 
-    public function collapsed(bool | Closure $condition = true, bool $shouldMakeComponentCollapsible = true): static
+    protected bool|Closure|null $shouldPersistCollapseState = null;
```
Comment (saade) [inline packages/infolists/src/Components/Concerns/CanBeCollapsed.php]:
```suggestion
    protected bool | Closure | null $shouldPersistCollapseState = null;
```

**Comment by zepfietje** [comment]
Could you undo the formatting changes by running Pint?

File: packages/infolists/src/Components/Concerns/CanBeCollapsed.php [hunk 4]
```diff
@@ -6,11 +6,13 @@
...
+
+    public function collapsed(bool|Closure $condition = true, bool $shouldMakeComponentCollapsible = true): static
```
Comment (saade) [inline packages/infolists/src/Components/Concerns/CanBeCollapsed.php]:
```suggestion
    public function collapsed(bool | Closure $condition = true, bool $shouldMakeComponentCollapsible = true): static
```

File: packages/infolists/src/Components/Concerns/CanBeCollapsed.php [hunk 5]
```diff
@@ -23,10 +25,10 @@ public function collapsed(bool | Closure $condition = true, bool $shouldMakeComp
 
     public function isCollapsed(): bool
     {
-        return (bool) $this->evaluate($this->isCollapsed);
+        return (bool)$this->evaluate($this->isCollapsed);
```
Comment (saade) [inline packages/infolists/src/Components/Concerns/CanBeCollapsed.php]:
```suggestion
        return (bool) $this->evaluate($this->isCollapsed);
```

File: packages/infolists/src/Components/Concerns/CanBeCollapsed.php [hunk 6]
```diff
@@ -23,10 +25,10 @@ public function collapsed(bool | Closure $condition = true, bool $shouldMakeComp
...
     }
 
-    public function collapsible(bool | Closure | null $condition = true): static
+    public function collapsible(bool|Closure|null $condition = true): static
```
Comment (saade) [inline packages/infolists/src/Components/Concerns/CanBeCollapsed.php]:
```suggestion
    public function collapsible(bool | Closure | null $condition = true): static
```

File: packages/infolists/src/Components/Concerns/CanBeCollapsed.php [hunk 7]
```diff
@@ -35,6 +37,18 @@ public function collapsible(bool | Closure | null $condition = true): static
 
     public function isCollapsible(): bool
     {
-        return (bool) $this->evaluate($this->isCollapsible);
+        return (bool)$this->evaluate($this->isCollapsible);
```
Comment (saade) [inline packages/infolists/src/Components/Concerns/CanBeCollapsed.php]:
```suggestion
        return (bool) $this->evaluate($this->isCollapsible);
```

File: packages/infolists/src/Components/Concerns/CanBeCollapsed.php [hunk 8]
```diff
@@ -35,6 +37,18 @@ public function collapsible(bool | Closure | null $condition = true): static
...
+    }
+
+    public function persistCollapseState(bool|Closure|null $condition = true): static
```
Comment (saade) [inline packages/infolists/src/Components/Concerns/CanBeCollapsed.php]:
```suggestion
    public function persistCollapseState(bool | Closure | null $condition = true): static
```

File: packages/infolists/src/Components/Concerns/CanBeCollapsed.php [hunk 9]
```diff
@@ -35,6 +37,18 @@ public function collapsible(bool | Closure | null $condition = true): static
...
+    {
+        $this->shouldPersistCollapseState = $condition;
+
+        return $this;
+    }
+
+    public function shouldPersistCollapseState(): bool
+    {
+        return (bool)($this->evaluate($this->shouldPersistCollapseState) ?? false);
```
Comment (saade) [inline packages/infolists/src/Components/Concerns/CanBeCollapsed.php]:
```suggestion
        return (bool) $this->evaluate($this->shouldPersistCollapseState);
```

File: packages/forms/resources/views/components/section.blade.php [hunk 10]
```diff
@@ -13,6 +13,7 @@
     :icon="$getIcon()"
     :icon-color="$getIconColor()"
     :icon-size="$getIconSize()"
+    :persistCollapseState="$shouldPersistCollapseState()"
```
Comment (zepfietje) [inline packages/forms/resources/views/components/section.blade.php]: Please use kebab case as elsewhere.

**Review by zepfietje — CHANGES_REQUESTED** [review]

**Comment by danharrin** [comment]
Hi, thanks for the PR. I think it's important to give you a bit of information as to why this PR has been tagged as "low priority". We really appreciate all contributions, and intend to visit this in the future, and hopefully merge it if it is a suitable feature for the framework.

However, reviewing PRs, and often rewriting them on behalf of the author to ensure they are consistent with the rest of the framework, consumes a lot of my time. At the moment we have a large backlog of bug reports that I am prioritising, since most of these affect functionality that worked in a previous version and are now impacting real projects. Hopefully you can see how that work comes first before introducing new features that people aren't using in projects yet.

Again, thank you for contributing and I'm looking forward to reviewing this PR when our backlog of bug reports has reduced.

**Comment by danharrin** [comment]
Thanks!

**Pull request merged**